### PR TITLE
rule_manager: generate default rules based on whether enable witness

### LINKED ISSUE #5220: Modify placement rule to support adding witness
## Development Task

On cloud env, TiKV is equipped with AWS EBS storage for a single node which provides the durability of 99.99%. Moreover, TiKV organizes 3 Raft replicas on top of that, which may be overkill for durability. To reduce cost, we can introduce a mechanism named "2 Replicas With 1 Log Only" instead of 3 replicas to save 30% storage resources and cut down about 100%(1 core) CPU usage empirically on the cloud. Compared with the original 3 replicas architecture, the "1 Log Only" replica stores raft logs but does not apply it, which still ensures data consistency through the Raft protocol. 

This issue records the progress in PD. Please refer to https://github.com/tikv/tikv/issues/12876 for the total progress.

### PR DESCRIPTION
**Pull request by nathan-edgar**
ref tikv/pd#5220

### What problem does this PR solve?

Issue Number: Close #5220 

### What is changed and how does it work?

```commit-message
rule_manager: generate default rules based on whether enable witness
```

### Check List

Tests

- Integration test
- Manual test
   - Setup cluster

Code changes

- None

Side effects

- None

Related changes

- None

### Release note

```release-note
None.
```

### CODE REVIEW

File: server/schedule/placement/rule_manager.go [hunk 1]
```diff
@@ -81,17 +81,43 @@ func (m *RuleManager) Initialize(maxReplica int, locationLabels []string) error
...
-			return err
+		var defaultRules []*Rule
+		if m.conf != nil && m.conf.IsWitnessAllowed() {
+			defaultRules = append(defaultRules,
+				[]*Rule{
+					{
+						GroupID:        "pd",
+						ID:             "default",
+						Role:           Voter,
+						Count:          maxReplica - 1,
```
[the remark this answers is not in the document]
Comment (nathan-edgar) [reply server/schedule/placement/rule_manager.go]: Change to round by 3, eg:

maxReplica 3 witness 1
maxReplica 5 witness 2
maxReplica 7 witness 2
maxReplica 4 witness 1
maxReplica 6 witness 2

[the remark this answers is not in the document]
Comment (nathan-edgar) [reply server/schedule/placement/rule_manager.go]: After discussion, instead directly divide by 2

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**